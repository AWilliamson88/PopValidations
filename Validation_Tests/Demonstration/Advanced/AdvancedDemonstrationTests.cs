using ApprovalTests;
using PopValidations_Tests.Demonstration.Advanced.Validators;
using PopValidations_Tests.TestHelpers;
using System.Threading.Tasks;
using Xunit;

namespace PopValidations_Tests.Demonstration.Advanced;

public class AdvancedDemonstrationTests
{
    [Fact]
    public async Task AdvancedValidator_ValidateWithSingleArtistAlbum_ToJson()
    {
        // Arrange
        var fakeAlbumDetailChecker = new FakeAlbumDetailsChecker();
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new AdvancedAlbumValidator(fakeAlbumDetailChecker));

        // Act
        var results = await runner.Validate(AdvancedTestData.SingleArtist());
        var json = JsonConverter.ToJson(results);

        // Assert
        Approvals.VerifyJson(json);
    }

    [Fact]
    public async Task AdvancedValidator_ValidateWithCollaborationAlbum_ToJson()
    {
        // Arrange
        var fakeAlbumDetailChecker = new FakeAlbumDetailsChecker();
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new AdvancedAlbumValidator(fakeAlbumDetailChecker));

        // Act
        var results = await runner.Validate(AdvancedTestData.Collaboration());
        var json = JsonConverter.ToJson(results);

        // Assert
        Approvals.VerifyJson(json);
    }

    [Fact]
    public void BasicValidator_Describe_ToJson()
    {
        // Arrange
        var fakeAlbumDetailChecker = new FakeAlbumDetailsChecker();
        var runner = ValidationRunnerHelper.BasicRunnerSetup(new AdvancedAlbumValidator(fakeAlbumDetailChecker));

        // Act
        var results = runner.Describe();
        var json = JsonConverter.ToJson(results);

        // Assert
        Approvals.VerifyJson(json);
    }
}

