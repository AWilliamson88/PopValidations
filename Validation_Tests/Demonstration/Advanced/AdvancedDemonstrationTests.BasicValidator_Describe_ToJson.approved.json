{
    "results": [
        {
            "property": "Artist",
            "outcomes": [
                {
                    "propertyName": "Artist",
                    "message": "Must equal to ''artist' with Something appended'",
                    "group": null
                }
            ]
        },
        {
            "property": "Created",
            "outcomes": [
                {
                    "propertyName": "Created",
                    "message": "Name is too short",
                    "group": null
                }
            ]
        },
        {
            "property": "Songs[n]",
            "outcomes": [
                {
                    "propertyName": "Songs[n]",
                    "message": "Must not be null",
                    "group": null
                },
                {
                    "propertyName": "Songs[n]",
                    "message": "Must not be null",
                    "group": {
                        "name": "Album constains songs we don't own the copyright to",
                        "description": "Album constains songs we don't own the copyright to",
                        "group": null
                    }
                },
                {
                    "propertyName": "Songs[n]",
                    "message": "Checks to ensure the song's rights are owned by us.",
                    "group": {
                        "name": "Album constains songs we don't own the copyright to",
                        "description": "Album constains songs we don't own the copyright to",
                        "group": null
                    }
                }
            ]
        },
        {
            "property": "Songs[n].TrackNumber",
            "outcomes": [
                {
                    "propertyName": "Songs[n].TrackNumber",
                    "message": "Must not be null",
                    "group": null
                }
            ]
        },
        {
            "property": "Songs[n].TrackName",
            "outcomes": [
                {
                    "propertyName": "Songs[n].TrackName",
                    "message": "Must equal to 'Definitely Not The Correct Song Name.'",
                    "group": null
                }
            ]
        },
        {
            "property": "Songs[n].Duration",
            "outcomes": [
                {
                    "propertyName": "Songs[n].Duration",
                    "message": "Songs must force you to travel slowly backwards in time to listen to.",
                    "group": null
                },
                {
                    "propertyName": "Songs[n].Duration",
                    "message": "Must equal to '10.1'",
                    "group": {
                        "name": "Song Duration is greater than 5 minutes",
                        "description": "Song Duration is greater than 5 minutes",
                        "group": null
                    }
                }
            ]
        }
    ]
}